src/pages/signinBlockstack.js

```diff
@@ -0,0 +1,259 @@
+      });
+  }
+
+  signinPublicKey(userData, publicKey, redirect) {
+    const user = { email: userData.email, publicKey };
+    this.signinPublicKeyPost(user, redirect)
+      .then(async response => {
+        if (response.redirect) {
+          const link = decryptContent(response.redirect, { privateKey: userData.appPrivateKey });
+          await Router.replaceRoute(link);
+        } else {
+          this.setState({ error: 'Failed to signin' });
+        }
+      })
+      .catch(error => {
+        this.setState({ error: `Failed to signin, ${error}` });
+      });
+  }
+
+  signinEmail(userData, publicKey, redirect) {
+    const user = { email: userData.email, publicKey };
+    this.signinEmailPost(user, `/blockstack/${encodeURIComponent(redirect)}`).catch(error => {
+      this.setState({ error: `Failed to signin, ${error}` });
+    });
+  }
+
+  checkUserExistence(email, publicKey) {
+    if (!isValidEmail(email)) return Promise.resolve(false);
+    return fetch(
+      `/api/users/exists-with-public-key?email=${encodeURIComponent(email)}&publicKey=${encodeURIComponent(publicKey)}`,
+    )
+      .then(checkResponseStatus)
+      .then(json => Boolean(json.exists));
+  }
+
+  signinPublicKeyPost(user, redirect) {
+    const websiteUrl = getWebsiteUrl();
+    return fetch('/api/users/signin-public-key', {
+      method: 'POST',
+      headers: {
+        ...this.addAuthTokenToHeader(),
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ user, redirect, websiteUrl }),
+    }).then(checkResponseStatus);
+  }
+
+  signinEmailPost(user, redirect) {
+    const websiteUrl = getWebsiteUrl();
+    return fetch('/api/users/signin', {
+      method: 'POST',
+      headers: {
+        ...this.addAuthTokenToHeader(),
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ user, redirect, websiteUrl }),
+    }).then(checkResponseStatus);
+  }
+
+  addAuthTokenToHeader(obj = {}) {
+    const accessToken = localStorage.getItem('accessToken');
+    if (!accessToken) return obj;
+    return {
+      Authorization: `Bearer ${accessToken}`,
+      ...obj,
+    };
+  }
+
+  render() {
+    const { userData, exists, existsEmail, loading, error } = this.state;
+    return (
+      <Page title="Blockstack Login Successful">
+        <Container pt={4} pb={6} px={2} background="linear-gradient(180deg, #EBF4FF, #FFFFFF)" textAlign="center">
+          <Flex justifyContent="center" mb={4}>
+            <Icon size="60" />
+          </Flex>
+          <H3 as="h1" fontWeight="800">
+            {userData && <>{userData.username}</>}
+            {loading && (
+              <>
+                <br />
+                ... stacking blocks ...
+              </>
+            )}
+            {!loading && exists && (
+              <>
+                <br />
+                You have already a profile. Signing in ...
```

Comment:
All text within this block should be localizable using `<FormatedMessage`

Reply:
FormattedMessage is used now.